### PR TITLE
added section about assumed security properties

### LINKED ISSUE #422: deal with deep hole of assumed security properties
> > -- The overall Section 12 seems silent on:
> >
> > o What is the implication of combining roles into a single entity as described in Section 3.4 and 6. Does this lack of separation present any additional issues?

[github said: https://github.com/ietf-rats-wg/architecture/issues/407]
==[ snip ]==
We need to acknowledge that there is a deep hole (not infinitely deep, but not trivial) where we need to look at integrity of all of the different platforms.
The way that the compositions are composed is a bit tricky, and the results are sometimes different than other people would naively assume.
Are there references here to other papers that we should include?
==[ snip ]==

I completely agree, that's why I mentioned it.  Composition is hard and can alter the security assumptions of the individual components and introduce emergent risk.

I don't think it can be solved generically.  However, we do need cautionary text here.  Perhaps:

NEW (roughly)

The RATS architecture allows for an entity to function in multiple roles (Section 6) and for composite devices (Section 3.3).  Implementers need to evaluate their designs to ensure that the assumed security properties of the individual components and roles still holds despite the lack of separation, and that emergent risk is not introduced.  The specifics of this evaluation will depend on the implementation and use case is out of scope for this document.

### PR DESCRIPTION
**Pull request by mcr**
insert text suggested by Roman about how deep down the security properties
close #422

### CODE REVIEW

**Review by nedmsmith — APPROVED** [review]

**Review by William-PanWei — APPROVED** [review]

File: draft-ietf-rats-architecture.md [hunk 1]
```diff
@@ -1451,6 +1451,11 @@ Security protections in RATS may be applied at different layers, whether by a co
 This architecture expects conceptual messages to be end-to-end protected based on the role interaction context.
 For example, if an Attester produces Evidence that is relayed through some other entity that doesn't implement the Attester or the intended Verifier roles, then the relaying entity should not expect to have access to the Evidence.
 
+The RATS architecture allows for an entity to function in multiple roles
+({{rolesentities}}) and for composite devices ({{compositedevice}}).
+Implementers need to evaluate their designs to ensure that the assumed security properties of the individual components and roles still holds despite the lack of separation, and that emergent risk is not introduced.
+The specifics of this evaluation will depend on the implementation and use case is out of scope for this document.
+
```
Comment (henkbirkholz) [inline draft-ietf-rats-architecture.md:1458]: Isolation mechanisms in software or hardware that separate Attesting Environments and Target Environments {{twotypes}} can support implementer's evalutaion and design decisions.

Comment (henkbirkholz) [inline draft-ietf-rats-architecture.md:1458]:
```suggestion
Isolation mechanisms in software or hardware that separate Attesting Environments and Target Environments {{twotypes}} can support implementer's evalutaion and design decisions.
```

File: draft-ietf-rats-architecture.md [hunk 2]
```diff
@@ -1451,6 +1451,11 @@ Security protections in RATS may be applied at different layers, whether by a co
 This architecture expects conceptual messages to be end-to-end protected based on the role interaction context.
 For example, if an Attester produces Evidence that is relayed through some other entity that doesn't implement the Attester or the intended Verifier roles, then the relaying entity should not expect to have access to the Evidence.
 
+The RATS architecture allows for an entity to function in multiple roles
+({{rolesentities}}) and for composite devices ({{compositedevice}}).
+Implementers need to evaluate their designs to ensure that the assumed security properties of the individual components and roles still holds despite the lack of separation, and that emergent risk is not introduced.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:1456]:
```suggestion
Implementers need to evaluate their designs to ensure that the assumed security properties of the individual components and roles still hold despite the lack of separation, and that emergent risk is not introduced.
```

File: draft-ietf-rats-architecture.md [hunk 3]
```diff
@@ -1451,6 +1451,11 @@ Security protections in RATS may be applied at different layers, whether by a co
...
+The specifics of this evaluation will depend on the implementation and use case is out of scope for this document.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:1457]:
```suggestion
The specifics of this evaluation will depend on the implementation and use case and hence is out of scope for this document.
```

**Pull request merged**